### PR TITLE
Doing the great thermostat renaming.

### PR DESCRIPTION
**Pull request by drempelg**
Doing a separate PR for just the renames of types.  This brings the cluster in line with the current version of the spec in master.

### CODE REVIEW

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter
Comment (drempelg) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:4048]: That's interesting, yeah it's not in there in the master spec.  Thermostat Running Mode is using SystemModeEnum in the spec, the tool still generated this though (the tool has some built in exceptions where neccesary to keep compatibility).

The XML in master also has this enum and while it's not directly referenced in the Thermostat Running Mode its an enum8, with the same values.  I gotta wonder if this was added to the XML but not changed in the spec?  Since the rest of the values in System Mode don't really make a long of sense for Running mode (which is used to indicate if Auto is heating or cooling at the moment).

Maybe a spec change is in order here?  I'll hunt down the git history of this enum.

Comment (drempelg) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:4048]: Looks like this is where it was added.

https://github.com/project-chip/connectedhomeip/commit/f2d408d26acfce2e050704ca6626bf5028072f8b

We are going to have to resolve this, what's the history on the spec?  Did the spec ever use ThermostatRunningMode?

**Review by robszewczyk — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:202]: This change from nullable to not nullable is a wire format compat break.  We can't make that change.  The spec needs to change here.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:214]: That's not really the right "default" value, but ok.  Followup: figure out why the tool is creating these not-really-correct values and fix it.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:216]: This should be of type OccupancyBitmap, in general.  I assume that change will happen in a separate PR?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:222]: The decimal defaults are in fact much more readable... Why that change?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:129]: This is removing a shipped attribute, no?  That's a compat break for APIs.  You can't do that.  Please leave it in, and we will need to figure out how to proceed (spec changes, a bunch of compat shims, something else), but we don't want to block this PR on that.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:229]: Again: The old default values were a lot more readable and more importantly matched the spec better, which made it easier to check that they are right.  Please undo these unnecessary changes.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:142]: This removal of min/max for _writable_ attributes is a breaking change: the attribute write machinery used to enforce the range automatically and now it won't.  The min/max need to stay for a bunch of these things.

And maybe file a followup issue to have that min/max inferred from the type in codegen, but that's going to take some work.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:249]: Again: why the gratuitous decimal-to-hex changes?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:291]: Not sure why the min/max noise is being added here (non-writable attributes).  It shouldn't be there.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:354]: The spec is buggy here.  It should have Y, not N, for the response to this command.  It even has prose talking about what status is sent in response! Please undo this change and get the spec fixed.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:369]: This command is not part of Matter and should not be here.  See "9.2.5. Element Requirements" in the device library spec.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:378]: This command is not part of Matter and should not be here.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
Changes requested because this has several compat breaks that we can't take (some just needing to be fixed in the XML, some maybe needing spec changes) and adds some non-Matter bits that should not be added.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter
Comment (hasty) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:4048]: No, I added it to #7848 so that it would be in sync with the SDK.

File: src/app/tests/suites/certification/Test_TC_TSTAT_2_1.yaml
Comment (drempelg) [reply src/app/tests/suites/certification/Test_TC_TSTAT_2_1.yaml:234]: Because the test was looking for an attribute that was marked D in the spec for 1.0.  We don't need it in the SDK, and it probably never should've been there, and it's no longer in the XML as a result.  So to pass the test we decided to remove it since it isn't actually a certification requirement.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:354]: @hasty can you make sure the spec gets the proper update?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:369]: hmmm...not sure how that got in there, that's from the updated incoming spec I believe.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:343]: @hasty you got this right?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:202]: @hasty

File: src/app/tests/suites/certification/Test_TC_TSTAT_2_1.yaml
Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_TSTAT_2_1.yaml:234]: We can't remove the attribute from the XML without breaking shipped public APIs or doing something complicated to not break them.  I strongly suggest leaving it in the XML for now for expediency and filing an issue on figuring out a way to remove it.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:216]: There's a ref for occupancy bitmap, but no definition.  @hasty this in your PR?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:129]: I brought this up with Cecil internally and she said we didn't need it.  It was deprecated in 1.0, probably should never have been used.  So long as the spec reserves the number, the SDK doesn't need it.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:229]: @hasty suggestions for the tool with the default values.  If we can keep them as decimal when the spec uses decimal we should

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:142]: @hasty didn't we remove these because we were informed that the temperature type already handled it?  I know we had them in a previous incarnation of this.

File: src/app/tests/suites/certification/Test_TC_TSTAT_2_1.yaml
Comment (drempelg) [reply src/app/tests/suites/certification/Test_TC_TSTAT_2_1.yaml:234]: ok, I'll restore the attribute and the unnecessary test since it could break API's.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:24]: I'll switch it back

Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml]: @hasty we will need to switch the spec back too correct?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (drempelg) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml]: @hasty seems we can't remove deprecated things from the XML, the need to remain.  Tool should include them even if deprecated :)

File: src/app/clusters/thermostat-server/thermostat-server.cpp
Comment (drempelg) [reply src/app/clusters/thermostat-server/thermostat-server.cpp:411]: https://github.com/project-chip/connectedhomeip/issues/30773

**Review by woody-apple — DISMISSED** [review]
Requesting changes, pending review from @bzbarsky-apple, this breaks public API as is.

**Comment by woody-apple** [comment]
> Requesting changes, pending review from @bzbarsky-apple, this breaks public API as is.

After further inspection, maybe it looks OK! Would like @bzbarsky-apple to review though.

File: src/app/clusters/thermostat-server/thermostat-server.cpp
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:100]:
```suggestion
            BitMask<RemoteSensingBitmap> valueRemoteSensing;
```

File: src/app/clusters/thermostat-server/thermostat-server.cpp
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:197]: This line should presumably not be here.

File: src/app/util/attribute-table.cpp
Comment (bzbarsky-apple) [inline src/app/util/attribute-table.cpp:1]: Why the random changes to this file?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml]: @hasty the spec _does_ have that....  In the device type.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml]: @hasty there is no way to express "null" as default in ZAP XML right now.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:230]: Why the random whitespace addition here?

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:283]: More invalid default values here...

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml]: @hasty They should not have done that.  I agree that this "oh, let's override in the device library" thing is pretty broken.  :(

**Review by bzbarsky-apple — APPROVED** [review]
Approved, modulo the remaining nits.

**Pull request merged**
